plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -77,42 +72,32 @@ def parse(cls, string: str) -> "CommandlineSpec":
              - set log distribution for scalars
                Eg: "int:log:4:1024"
         """
-        available_modifiers = {"log", "float", "int", "str"}
-        colon_split = string.split(":")
-        modifiers = set(
-            itertools.takewhile(available_modifiers.__contains__, colon_split)
-        )
-        remain = colon_split[len(modifiers) :]
-        casts = list(modifiers - {"log"})
-        if len(remain) not in {1, 2}:
-            raise ValueError(
-                "Can't interpret non-speficiations: {}.\nthis needs to be "
-                "either colon or coma-separated values".format(":".join(remain))
-            )
-        if len(casts) > 1:
-            raise ValueError(f"Inconsistent specifications: {casts}")
-        if len(remain) == 1:  # choice argument
-            cast = casts[0] if casts else "str"
-            options = remain[0].split(",")
-            if len(options) < 2:
+        if override.is_choice_sweep():
+            choices = override.value().list
+            if len(choices) < 2:
                 raise ValueError("At least 2 options are required")
-            if not casts:
-                try:  # default to float if possible and no spec provided
-                    _ = [float(x) for x in options]
-                    cast = "float"
-                except ValueError:
-                    pass
-            return cls(options=options, cast=cast)
-        # bounded argument
-        bounds: Tuple[float, float] = tuple(float(x) for x in remain)  # type: ignore
-        cast = casts[0] if casts else "float"
-        return cls(bounds=bounds, cast=cast, log="log" in modifiers)
+            return cls(options=choices)
+        elif override.is_range_sweep():
+            available_tags = {"log", "step"}
+            # the only valid tag for now is log and step, is step is present, we will set the value
+            value = override.value()
+            tags = value.tags
+            for t in tags:
+                if t not in available_tags:
+                    raise ValueError(
+                        f"Tag {t} not supported. Only {available_tags} are supported."
+                    )
+            step = float(value.step) if "step" in tags else None
+            return cls(bounds=(value.start, value.stop), log=("log" in tags), step=step)
+        elif override.value_type == ValueType.ELEMENT:
+            return override.value()
+        else:
+            raise ValueError(f"Unsupported Override: {override}")
 
 
 # pylint: disable=too-many-branches
 def make_nevergrad_parameter(description: Any) -> Any:
     """Returns a Nevergrad parameter from a definition string or object.
```

Comment:
The input for this function should be an Override object.
It should probably be re-written from scratch.